src/OrchardCore.Modules/OrchardCore.Users/Services/EmptyRoleStore.cs

```diff
@@ -0,0 +1,63 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using OrchardCore.Security;
+
+namespace OrchardCore.Users.Services;
+
+public class EmptyRoleStore : IRoleStore<IRole>
```

Comment:
Can we name `NullRoleStore` and move it into `OC.Roles`, then the registration for the services can be done in extension methods

Reply:
We could rename it. I don't think we should put it in roles. I think it should even be internal to Users.

But either way, I think this fixes the problem on hand but create another problem

Reply:
It's quite similar to `NullLocalizer` to make a default `IStringLocalizer` when the localization feature is disabled

Reply:
agreed. In this case we are dealing with claims where are managed at the Role level currently. if we don't want to require roles, then we need a service someone that would provide claims/permission when the user first login "including admin" otherwise the site won't be usable

Reply:
I agree with `NullRoleStore` name. It doesn't return an empty result but a null result.